### ISSUE TITLE
supabase function 204 fails with type error

### ISSUE THREAD

**Issue by johanneswuerbach** [supabase/cli#1071]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Trying to return an http 204, fails with a 500 and a TypeError in the console.

```
TypeError: Response with null body status cannot have body
    at initializeAResponse (ext:deno_fetch/23_response.js:202:13)
    at new Response (ext:deno_fetch/23_response.js:337:5)
    at UserWorker.fetch (ext:sb_user_workers/user_workers.js:52:16)
    at async Server.<anonymous> (file:///home/deno/main/index.ts:102:16)
    at async Server.#respond (https://deno.land/std@0.182.0/http/server.ts:220:24)
```

## To Reproduce

Create function with the following:

```ts
import { serve } from 'std/server'

serve(async (req: Request) => {
  return new Response(null, { status: 204 })
})
```

```json
{
  "imports": {
    "std/server": "https://deno.land/std@0.186.0/http/server.ts"
  }
}
```

## Expected behavior

A 204 response without a response body.

## System information

- OS: macOS
- Supabase CLI 1.54.5

**Comment by laktek** [comment]
Thanks for opening a ticket @johanneswuerbach - will take a look.

**Comment by laktek** [comment]
We had to revert this fix as it was breaking other streaming responses (for example, Server-Sent Events)

**Comment by laktek** [comment]
Fixed in CLI 1.61.3
